app/bundles/LeadBundle/LeadEvents.php

```diff
@@ -379,6 +379,46 @@
     const DEVICE_POST_DELETE = 'mautic.lead_device_post_delete';
 
     /**
+     * The mautic.lead_device_pre_save event is dispatched right before a lead device is persisted.
+     *
+     * The event listener receives a
+     * Mautic\LeadBundle\Event\LeadEvent instance.
+     *
+     * @var string
+     */
+    const TAG_PRE_SAVE = 'mautic.lead_device_pre_save';
```

Comment:
Be sure to update all of these event values or they'll collide with DEVICE_PRE_SAVE.